dashboard/src/app/dashboard/[dashboardId]/web-vitals/interactiveWebVitalsChartSection.tsx

```diff
@@ -0,0 +1,230 @@
+'use client';
+import { use, useState } from 'react';
+import { cn } from '@/lib/utils';
+import SummaryCardsSection, { SummaryCardData } from '@/components/dashboard/SummaryCardsSection';
+import CoreWebVitalBar from '@/components/dashboard/CoreWebVitalBar';
+import { CoreWebVitalName, CoreWebVitalsSummary } from '@/entities/webVitals';
+import MultiSeriesChart from '@/components/MultiSeriesChart';
+import type { MultiSeriesConfig } from '@/components/MultiSeriesChart';
+import { useTimeRangeContext } from '@/contexts/TimeRangeContextProvider';
+import { CoreWebVitalsSeries } from '@/presenters/toMultiLine';
+import { formatShortFromMilliseconds, formatCompactFromMilliseconds } from '@/utils/dateFormatters';
+import { formatCWV, getCwvStatusColor } from '@/utils/formatters';
+import { CWV_THRESHOLDS } from '@/constants/coreWebVitals';
+import MetricInfo from '@/app/dashboard/[dashboardId]/web-vitals/MetricInfo';
+
+type Props = {
+  summaryPromise: Promise<CoreWebVitalsSummary>;
+  seriesPromise: Promise<CoreWebVitalsSeries>;
+};
+
+export default function InteractiveWebVitalsChartSection({ summaryPromise, seriesPromise }: Props) {
+  const summary = use(summaryPromise);
+  const { granularity } = useTimeRangeContext();
+  const [active, setActive] = useState<CoreWebVitalName>('CLS');
+  const seriesByMetric = use(seriesPromise);
+
+  function formatCardValue(metric: CoreWebVitalName, value: number | null): React.ReactNode {
+    if (value === null) return '—';
+    const colorStyle: React.CSSProperties = { color: getCwvStatusColor(metric, value) };
+
+    const display = metric === 'CLS' ? value.toFixed(3) : formatCompactFromMilliseconds(value);
+    return <span style={colorStyle}>{display}</span>;
+  }
```

Comment:
Suggestion: Move this to outside of component function to prevent recomputing it on every re-render.